src/app/Shared/Services/Api.service.tsx

```diff
@@ -90,36 +90,45 @@ export class ApiService {
     .pipe(concatMap(resp => from(resp.json())));
     const getDashboardURL = fromFetch(`${this.login.authority}/api/v1/grafana_dashboard_url`)
     .pipe(concatMap(resp => from(resp.json())));
-
-    fromFetch(`${this.login.authority}/health`)
+    const health = fromFetch(`${this.login.authority}/health`)
       .pipe(
-        concatMap(resp => from(resp.json())),
-        concatMap((jsonResp: any) => {
-          if (!!jsonResp.cryostatVersion && jsonResp.dashboardAvailable && jsonResp.datasourceAvailable) {
-            return forkJoin([of(jsonResp.cryostatVersion), getDatasourceURL, getDashboardURL]);
-          } else {
-            const missing: string[] = [];
-            if (!jsonResp.cryostatVersion) {
-              missing.push('Cryostat version');
-            }
-            if (!jsonResp.dashboardAvailable) {
-              missing.push('dashboard URL');
-            }
-            if (!jsonResp.datasourceAvailable) {
-              missing.push('datasource URL');
-            }
-            const message = missing.join(', ') + ' unavailable';
-            return throwError(() => new Error(message));
-          }}))
+        tap((resp: Response) => {
+          if (!resp.ok) {
+            window.console.error(resp);
+            this.notifications.danger('API /health request failed', resp.statusText);
+          }
+        }),
+        concatMap((resp: Response) => from(resp.json())),
+        shareReplay(),
```

Comment:
Is the `shareReplay()` needed here because the subscriptions on lines 104/108 subscribe to `health` after the first response gets emitted?

Reply:
So if I'm understanding correctly, the separate subscriptions on lines 105 and 109 both receive and operate on the same cached `Observable` response JSON created on line 101, which is why they don't need to do the `(resp: Response) => from(resp.json())` conversion? And this allows you to extract the version information separately from the datasource/dashboard handling while limiting side effects (i.e. the original `fromFetch` API call) to only the amount needed?

Reply:
You got it!